src/pset/elip100.rs

```diff
@@ -46,13 +49,41 @@ impl PartiallySignedTransaction {
         &self,
         asset_id: AssetId,
     ) -> Option<Result<AssetMetadata, encode::Error>> {
-        let key = prop_key(&asset_id);
+        let key = prop_key(&asset_id, PSBT_ELEMENTS_HWW_GLOBAL_ASSET_METADATA);
 
         self.global
             .proprietary
             .get(&key)
             .map(|data| AssetMetadata::deserialize(data))
     }
+
+    /// Add token information to the PSET, returns None if it wasn't present or Some with the old
+    /// data if already in the PSET
```

Comment:
Should we mention in the doc that you need also to add the relative contract metadata?

Reply:
Or should we change `add_asset_metadata` to optionally accept `TokenMetadata`?

Reply:
Perhaps both.

In this PR I wanted to be the least disruptive possible, and just add a way to add/get token metadata in the PSET.

We could either:
* add an alternative method that sets asset and (optionally) token metadata at once (verifying their consistency and contract commitment)
* change `add_asset_metadata` to accept `Option<TokenMetadata>`

But those require a bit of discussion, and it might take time.

A viable approach could be to go with this PR and later add a "safer to use" method with a separate PR.
Then we can deprecate the `add/get_asset/token_metadata`, or keep them if we feel to. 
In this way we can think a bit more about the interface we want to expose, while not blocking who needs to use this new fields.